crates/light-client-prover/src/circuit.rs

```diff
@@ -94,27 +104,65 @@ pub fn run_circuit<DaV: DaVerifier, G: ZkvmGuest>(
                         let journal =
                             G::extract_raw_output(&proof).expect("DaData proofs must be valid");
                         // TODO: select output version based on the spec
-                        let batch_proof_output: BatchProofCircuitOutput<DaV::Spec, [u8; 32]> =
-                            match G::verify_and_extract_output(
-                                &journal,
-                                &batch_proof_method_id.into(),
-                            ) {
-                                Ok(output) => output,
-                                Err(_) => continue,
+
+                        let batch_proof_method_id = if batch_proof_method_ids.len() == 1 {
+                            // Check if last l2 height is greater than or equal to the only batch proof method id activation height
+                            if last_l2_height >= batch_proof_method_ids[0].0 {
+                                batch_proof_method_ids[0].1
+                            } else {
+                                // If not continue to the next blob
+                                continue 'blobs;
+                            }
+                        } else {
+                            let idx = match batch_proof_method_ids
+                                // Returns err and the index to be inserted, which is the index of the first element greater than the key
+                                // That is why we need to subtract 1 to get the last element smaller than the key
+                                .binary_search_by_key(&last_l2_height, |(height, _)| *height)
```

Comment:
you shouldn't be searching by last_l2_height, this is light client proofs last l2 height

extract output first, then use that last_l2 height and use verify.